packages/perspective-viewer-d3fc/test/js/axisLabel.spec.ts

```diff
@@ -0,0 +1,335 @@
+// ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
+// ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+// ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
+// ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
+// ┃ This file is part of the Perspective library, distributed under the terms ┃
+// ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
+// ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+import { expect, test } from "@finos/perspective-test";
+
+test.beforeEach(async ({ page }) => {
+    await page.goto("/tools/perspective-test/src/html/basic-test.html");
+    await page.evaluate(async () => {
+        while (!window["__TEST_PERSPECTIVE_READY__"]) {
+            await new Promise((x) => setTimeout(x, 10));
+        }
+    });
+
+    await page.evaluate(async () => {
+        let viewer = document.querySelector("perspective-viewer");
+        await viewer!.restore({
+            plugin: "Datagrid",
+            group_by: ["Order Date", "Profit"],
+            x_axis: ["Row ID"],
+        });
+
+        return await viewer.save();
+    });
+
+    await page.click("#settings_button");
+    await page.click(".plugin-select-item");
+});
+
+function confirmDateData(dateValues: any[]) {
+    // expect the first to be a date.
+    expect(!isNaN(Date.parse(dateValues[0]))).toEqual(true);
+
+    // expect any random index of axisLabels to also be a date.
+    let len = dateValues.length;
+    const index = Math.floor(Math.random() * len);
+
+    expect(!isNaN(Date.parse(dateValues[index]))).toEqual(true);
+
+    // expect the last value to be a date.
+
+    expect(!isNaN(Date.parse(dateValues[len - 1]))).toEqual(true);
+
+    /**
+     * To confirm that the date string is not in epoch format.
+     * check if the entire string is numeric.
+     * if it is, confirm that the length is not 10 or 13.
+     * if it is not, return true.
+     */
+    let isNumericOnly = /^\d+$/.test(dateValues[index]);
+
+    expect(isNumericOnly).toEqual(false);
+}
+
+test.describe("Axis Values With Grouped Data With A Date Field In The Group", () => {
+    test("X Bar y-axis label with grouped data", async ({ page }) => {
+        await page.click('div[data-plugin="X Bar"]');
+        await page.waitForSelector("perspective-viewer");
+
+        const dateValues = await page.evaluate(async () => {
+            let viewer = document.querySelector("perspective-viewer");
+
+            if (!viewer) return [];
+
+            const plugin_element = viewer.querySelector(
+                `perspective-viewer-d3fc-xbar`
+            );
+
+            if (!plugin_element) return [];
+
+            const shadowRoot = plugin_element.shadowRoot;
+            const elements = shadowRoot.querySelectorAll(
+                "div d3fc-group d3fc-svg.y-axis.left-axis svg, g.group"
+            );
```

Comment:
Why does this select `div d3fc-group d3fc-svg.y-axis.left-axis svg` only to drop it on line 85? I think this was intended to be `div d3fc-group d3fc-svg.y-axis.left-axis svg g.group` (without a comma), but because you only take the "last" group the extra selects does not affect the test outcome. It does make this code hard to read though. You could also just combine this entire section into the selector `div d3fc-group d3fc-svg.y-axis.left-axis svg g.group:last-child g.tick text`.